src/analysisd/labels.c

```diff
@@ -70,12 +70,12 @@ wlabel_t* labels_find(const Eventinfo *lf) {
     w_mutex_lock(&label_mutex);
```

Comment:
I suggest to use **lock** and **unlock** atomically around the hash methods, avoiding locking and unlocking in different scopes because it can be difficult to sustain. i.e:
```
w_mutex_lock(&label_mutex);
data = (wlabel_data_t*)OSHash_Get(label_cache, lf->agent_id)
w_mutex_unlock(&label_mutex);
if(!data){
```
...
```
w_mutex_lock(&label_mutex);
int hash_result  = OSHash_Add(label_cache, lf->agent_id, data) 
w_mutex_unlock(&label_mutex);
if(hash_result  != 2) {
```
...

Reply:
@pereyra-m @palaciosjeremias 
Please, be as picky as possible and review the method again. After discussing with Victor I decided to fully re-implement it.